app/src/gui/components/record/relationships/field_level_links/datagrid.tsx

```diff
@@ -190,122 +209,105 @@ export default function DataGridFieldLinksComponent(
-      headerName: 'Relationship',
-      headerClassName: 'faims-record-link--header',
-      minWidth: 200,
-      flex: 0.2,
-      valueGetter: (params: gridParamsDataType) => params.value[1],
-    },
-    {
-      field: 'record',
-      headerName: 'Record',
-      headerClassName: 'faims-record-link--header',
-      minWidth: 200,
-      flex: 0.4,
-      valueGetter: (params: GridCellParams) =>
-        params.row.type + ' ' + params.row.hrid,
-      renderCell: (params: GridCellParams) =>
-        recordDisplay(
-          props.record_id,
-          params.row.record_id,
-          params.row.type,
-          params.row.hrid,
-          params.row.route,
-          params.row.deleted
-        ),
-    },
 
-    {
-      field: 'lastUpdatedBy',
-      headerName: 'Last Updated By',
-      headerClassName: 'faims-record-link--header',
-      minWidth: 150,
-      flex: 0.2,
-      valueGetter: (params: GridCellParams) => params.row.lastUpdatedBy,
-    },
-  ];
-  // for read ONLY
+  function ChildRecordDisplay(props: {
+    current_record_id: RecordID;
+    child_record: RecordLinkProps;
+  }) {
+    const [recordData, setRecordData] = useState<
+      | {
+          latest_record: null;
+          revision_id: undefined;
+        }
+      | {
+          latest_record: Record | null;
+          revision_id: string;
+        }
+    >();
 
-  // BBS 20221117 checking on empty label to toggle. Label is set in src/gui/fields/RelatedRecordSelector.tsx
-  if (
-    props.relation_type === 'Child' &&
-    props.disabled !== true &&
-    props.relation_preferred_label !== ''
-  )
-    columns.push({
-      field: 'preferred',
-      headerName: 'Make ' + props.relation_preferred_label,
-      headerClassName: 'faims-record-link--header',
-      minWidth: 100,
-      flex: 0.2,
-      align: 'center',
-      headerAlign: 'center',
-      valueGetter: (params: GridCellParams) =>
-        params.row.relation_preferred ?? false,
-      renderCell: (params: GridCellParams) => (
-        <Checkbox
-          checked={params.row.value}
-          disabled={
-            props.preferred !== undefined &&
-            props.preferred !== null &&
-            props.preferred !== params.row.record_id
-              ? true
-              : false
-          }
-          onChange={(event: any) => {
-            if (props.handleMakePreferred !== undefined)
-              props.handleMakePreferred(
-                params.row.record_id,
-                event.target.checked
+    const [displayFields, setDisplayFields] = useState<Array<string>>([]);
+
+    useEffect(() => {
+      const fn = async () => {
+        // get the record so we can display some fields
+        const rd = await getRecordInformation(props.child_record);
+        setRecordData(rd);
+
+        if (uiSpec && rd?.latest_record)
+          setDisplayFields(getSummaryFields(uiSpec, rd.latest_record.type));
+      };
+      fn();
+    }, [props.child_record]);
```

Comment:
Might be worth starting to use React Query instead of useEffects to manage these async fetches. 

This merges the useState, the useEffect etc into one operation which includes loading, error by default so a bit neater/nicer.

Reply:
Removed the need for a query here by getting the full record data before we get here.